src/getAvailable.js

```diff
@@ -24,17 +24,10 @@ const getAvailable = async (req, res) => {
     utils.verbose(`No narratives available for ${source.name}`);
   }
 
-  // XXX TODO: This is bad and should go away by refactoring second tree
-  // enumeration into our source classes, potentially as part of the
-  // availableDatasets() method.
-  //   -trs, 3 Oct 2019
-  const secondTreeOptions = (dataset) =>
-    (global.availableDatasets.secondTreeOptions[source.name] || {})[dataset] || [];
-
   return res.json({
-    datasets: datasets.map((path) => ({
+    datasets: datasets.map((path, idx) => ({
       request: joinPartsIntoPrefix({source, prefixParts: [path]}),
-      secondTreeOptions: secondTreeOptions(path),
+      secondTreeOptions: secondTreeOptions[idx],
```

Comment:
Shared indexing into dual arrays generally indicates that a better data representation/structure is warranted. It is all too easy for such dual arrays to become out of sync and produce bugs. Two thoughts:

1. Instead of `secondTreeOptions` switching from a function that takes a path to an array which takes an index, it could be a dictionary/object which takes a path.

2. Instead of `secondTreeOptions` switching from a function at all, how about making it a method that takes a path on the source object? The change to make `availableDatasets` return two lists would go away and it can be left simpler.

Reply:
I went with option 2 in a new, fixup commit.